treenodes.md

```diff
@@ -0,0 +1,182 @@
+# Varieties of Tree Nodes
```

Comment:
isn't it in a docs subdirectory?

... doh. I totally created one, and then didn't cd to it before creating the file.